metricflow/specs.py

```diff
@@ -107,6 +111,24 @@ def qualified_name(self) -> str:
         raise NotImplementedError()
 
 
+@dataclass(frozen=True)
+class MetadataSpec(InstanceSpec):
```

Comment:
This looks fine for now, but there will likely be different types of metadata later on so there might need to be subclasses of this. I'm also thinking whether to make this `Visitable` (also for later).